### PR TITLE
fix a bug in resolving registry dependencies with alternative manifest versions

### PR DESCRIPTION
**Pull request by tomerd**
motivation: fix an existing issue with registry dependencies that can lead to resolution failure

changes:
* improve the placeholder manifest lookup of registry based dependencies that have alternative manifests
* add test

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/Workspace/RegistryPackageContainer.swift [hunk 1]
```diff
@@ -173,8 +173,20 @@ public class RegistryPackageContainer: PackageContainer {
                                 return completion(.failure(error))
                             case .success(let manifestContent):
                                 do {
+                                    // find the fake manifest so we can replace it with the real manifest content
+                                    guard let placeholderManifestFileName = try fileSystem.getDirectoryContents(.root).first(where: { file in
+                                        if file == Manifest.basename + "@swift-\(preferredToolsVersion).swift" {
+                                            return true
+                                        } else if preferredToolsVersion.patch == 0, file == Manifest.basename + "@swift-\(preferredToolsVersion.major).\(preferredToolsVersion.minor).swift" {
+                                            return true
+                                        } else  {
```
Comment (yim-lee) [inline Sources/Workspace/RegistryPackageContainer.swift:182]: nit: extra space

**Review by yim-lee — APPROVED** [review]

File: Sources/Workspace/RegistryPackageContainer.swift (hunk 1 above)
Comment (tomerd) [inline Sources/Workspace/RegistryPackageContainer.swift:182]:
```suggestion
                                        } else {
```

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**